WordPress/src/main/java/org/wordpress/android/ui/blaze/ui/blazeoverlay/BlazeOverlayFragment.kt

```diff
@@ -0,0 +1,282 @@
+import androidx.compose.foundation.layout.padding
+import androidx.compose.foundation.layout.size
+import androidx.compose.foundation.layout.width
+import androidx.compose.foundation.layout.wrapContentHeight
+import androidx.compose.foundation.lazy.LazyColumn
+import androidx.compose.foundation.lazy.items
+import androidx.compose.foundation.shape.RoundedCornerShape
+import androidx.compose.material.MaterialTheme
+import androidx.compose.material.Text
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.getValue
+import androidx.compose.runtime.livedata.observeAsState
+import androidx.compose.ui.Alignment
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.draw.clip
+import androidx.compose.ui.graphics.Color
+import androidx.compose.ui.platform.ComposeView
+import androidx.compose.ui.res.dimensionResource
+import androidx.compose.ui.res.painterResource
+import androidx.compose.ui.res.stringResource
+import androidx.compose.ui.text.font.FontWeight
+import androidx.compose.ui.text.style.TextAlign
+import androidx.compose.ui.tooling.preview.Preview
+import androidx.compose.ui.unit.dp
+import androidx.constraintlayout.compose.ConstraintLayout
+import androidx.constraintlayout.compose.Dimension
+import androidx.fragment.app.Fragment
+import androidx.fragment.app.activityViewModels
+import coil.compose.rememberImagePainter
+import dagger.hilt.android.AndroidEntryPoint
+import org.wordpress.android.R
+import org.wordpress.android.ui.blaze.BlazeUiState
+import org.wordpress.android.ui.blaze.PostUIModel
+import org.wordpress.android.ui.compose.components.Drawable
+import org.wordpress.android.ui.compose.components.ImageButton
+import org.wordpress.android.ui.compose.theme.AppColor
+import org.wordpress.android.ui.compose.theme.AppTheme
+import org.wordpress.android.ui.compose.unit.FontSize
+import org.wordpress.android.ui.compose.unit.Margin
+import org.wordpress.android.ui.compose.utils.LocaleAwareComposable
+import org.wordpress.android.ui.utils.UiString
+import org.wordpress.android.util.LocaleManager
+
+@AndroidEntryPoint
+class BlazeOverlayFragment : Fragment() {
+    companion object {
+        fun newInstance() = BlazeOverlayFragment()
+    }
+
+    private val viewModel: BlazeViewModel by activityViewModels()
+
+    override fun onCreateView(
+        inflater: LayoutInflater, container: ViewGroup?,
+        savedInstanceState: Bundle?
+    ): View = ComposeView(requireContext()).apply {
+        setContent {
+            AppTheme {
+                val userLanguage by viewModel.refreshAppLanguage.observeAsState("")
+                val postModel by viewModel.promoteUiState.observeAsState(BlazeUiState.PromoteScreen.Site)
+                LocaleAwareComposable(
+                    locale = LocaleManager.languageLocale(userLanguage),
+                    onLocaleChange = viewModel::setAppLanguage
+                ) {
+                    viewModel.trackOverlayDisplayed()
```

Comment:
Not sure how familiar you are with _recomposition_, there's the gist of it and what it entails for development explained in the [Compose Mental Model](https://developer.android.com/jetpack/compose/mental-model) article. What I'm trying to say is that if I'd have written this code I would have avoided to call tracking methods from Compose UI 👍.

There might be no need to change anything here; only if you want. Wdyt of moving the tracking to viewModel's start or the fragment init method?! Would that work in this case? That way if someone starting with Compose tries to copy from here they won't leave a door open to easily avoidable issues.

Reply:
Hey @ovitrif 

Thanks for that suggestion and information. I have gone through the above link, I see what you are talking about. 
I have moved the tracking logic to the ViewModel with the commit 9f7d6693e0bff673c3ba0449d58f9faab846574b.